gpytorch/kernels/linear_kernel.py

```diff
@@ -67,6 +67,12 @@ def __init__(
             self.register_prior("variance_prior", variance_prior, lambda m: m.variance, lambda m, v: m._set_variance(v))
 
         self.register_constraint("raw_variance", variance_constraint)
+        self.register_parameter(
```

Comment:
Should we be concerned about backward compatibility for loading old state dicts? Or are we ok breaking this?

Reply:
Fair question. I think a breaking change is probably fine here?

Reply:
Since we initialize to zero it should probably be fine?